Assets/Scripts/Crow/CrowManager.cs

```diff
@@ -0,0 +1,107 @@
+    private const int endXPos = -5; //Offscreen X location 'behind' the player
+    private int zPos = 0; //Initial z-axis location before updating with random value
+
+    /// <summary>
+    /// Spawn an initial crow at a location
+    /// </summary>
+    private void Start()
+    {
+        SpawnCrow();
+    }
+
+    /// <summary>
+    /// Instantiates a Crow gameobject at a position offscreen (x-axis) at a random point on the z-axis
+    /// Start Coroutine to move the crow
+    /// </summary>
+    private void SpawnCrow()
+    {
+        zPos = GetRandomZPos();
+        Vector3 spawnPosition = new Vector3(startXPos, 0f, zPos);
+        GameObject spawnedCrow = Instantiate(crowModel, spawnPosition, Quaternion.identity);
+        StartCoroutine(MoveCrow(spawnedCrow));
+    }
+
+    /// <summary>
+    /// Moves a crow gameobject along the x-axis towards an end point
+    /// Waits before moving the crow again at a new random z-axis position
+    /// </summary>
+    /// <param name="crow">Instantiated crow gameobject spawned in the scene</param>
+    /// <returns>WaitForSeconds to delay Crow respawning</returns>
+    private IEnumerator MoveCrow(GameObject crow)
+    {
+        //Show an alert when the crow spawns
+        StartCoroutine(ShowAlert()); 
+
+        //Set values for time and start/end positions
+        float elapsedTime = 0f;
+        Vector3 startPos = crow.transform.position;
+        Vector3 endPos = new Vector3(endXPos, startPos.y, startPos.z);
+
+        //Move the crow over time from a start to end point
+        while (elapsedTime < travelTime)
+        {
+            crow.transform.position = Vector3.Lerp(startPos, endPos, elapsedTime / travelTime);
+            elapsedTime += Time.deltaTime;
+            yield return null;
+        }
+
+        // Move to a new position offscreen and wait (after reaching the end)
+        crow.transform.position = new Vector3(30f, 0f, 30f); //Offscreen location
+        yield return new WaitForSeconds(Random.Range(minSpawnDelay, maxSpawnDelay)); //Wait a random amount before 'respawning'
+
+        //Get a new z-axiz position for the crow to move along
+        zPos = GetRandomZPos(); 
+
+        // Move crow to new start position and repeat movement coroutine
+        crow.transform.position = new Vector3(startXPos, 0f, zPos);
+        StartCoroutine(MoveCrow(crow));
+    }
+
+    /// <summary>
+    /// Toggles the alert above the player to be active for half the crow travel time
+    /// </summary>
+    /// <returns>WaitForSeconds before hiding gameobject again</returns>
+    private IEnumerator ShowAlert()
+    {
+        playerAlert.SetActive(true);
+        yield return new WaitForSeconds(travelTime / 2f);
+        playerAlert.SetActive(false);
+    }
+
+    /// <summary>
+    /// Sets a random z-axis starting position within player move space
+    /// </summary>
+    /// <returns>z position for new crow spawn location</returns>
+    private int GetRandomZPos() => Random.Range(1, 9);
```

Comment:
this will need to be changed to use the grid width value, instead of specifying 9